### PR TITLE
MBL-766: FrequentlyAskedQuestionsViewHolderViewModel.kt to Jetpack ViewModel and RxJava2.2

### PR DESCRIPTION
**Pull request by leighdouglas**
# 📲 What

Migration to rxjava 2.2 and jetpacks viewmodel

# 🤔 Why

[Technical Details](https://kickstarter.atlassian.net/wiki/spaces/NT/pages/2350317579/ViewModel+Tech+Debt+Epic#Technical-details)

# 👀 See

No user facing changes

# 📋 QA

- Find a project with FAQ questions (I used "Campster 2 the portable chair for everyone everywhere")
- Make sure you can view the FAQ tab, that there are no crashes, formatting is okay, scrolling works, etc.
- 
# Story 📖

[MBL-766: FrequentlyAskedQuestionsViewHolderViewModel.kt to Jetpack ViewModel and RxJava2.2](https://kickstarter.atlassian.net/browse/MBL-766)

### CODE REVIEW

**Review by mtgriego — CHANGES_REQUESTED** [review]
Looks good other than the disposables in `FrequentlyAskedQuestionsViewHolder.kt` dont seem to be cleared anywhere

**Comment by leighdouglas** [comment]
> Looks good other than the disposables in `FrequentlyAskedQuestionsViewHolder.kt` dont seem to be cleared anywhere

Done! Good catch :)

**Review by mtgriego — APPROVED** [review]

File: app/src/main/java/com/kickstarter/ui/viewholders/FrequentlyAskedQuestionsViewHolder.kt [hunk 1]
```diff
@@ -1,35 +1,43 @@
...
+            .observeOn(AndroidSchedulers.mainThread())
             .subscribe { binding.questionAnswerLayout.setLastedUpdateDate(it) }
+            .addToDisposable(disposables)
     }
 
     override fun bindData(data: Any?) {
         this.viewModel.inputs.configureWith(data as ProjectFaq)
     }
+
+    override fun destroy() {
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/viewholders/FrequentlyAskedQuestionsViewHolder.kt:39]: When it's a ViewHolder, you need to make sure the Adapter is calling destroy to clean the disposables, In this case `FrequentlyAskedQuestionsAdapter` inherits from `KSListAdapter` you need to add on the `onViewDetachedFromWindow
Screenshot 2023-06-29 at 2 02 50 PM
` the active call to `destroy`,  that before was handled on `ActivityViewModel`
Screenshot 2023-06-29 at 2 02 50 PM

File: app/src/main/java/com/kickstarter/viewmodels/FrequentlyAskedQuestionsViewHolderViewModel.kt [hunk 2]
```diff
@@ -25,31 +25,39 @@ interface FrequentlyAskedQuestionsViewHolderViewModel {
         fun updatedDate(): Observable<String>
     }
 
-    class ViewModel(@NonNull val environment: Environment) :
-        ActivityViewModel<FrequentlyAskedQuestionsViewHolder>(environment), Inputs, Outputs {
+    class FrequentlyAskedQuestionsViewHolderViewModel() : ViewModel(), Inputs, Outputs {
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/viewmodels/FrequentlyAskedQuestionsViewHolderViewModel.kt:28]: This inheritance works only for ViewModels on `Activity/Fragments`. We do not build this viewModel using a factory, and if take a look into the `ViewModel()` constructor source code should never be used without it. 
Screenshot 2023-06-29 at 2 07 14 PM

Remove this inheritance, instead create a new input method, that will be called from the `onDestroy()` on the viewHolder. There is an example in 
Screenshot 2023-06-29 at 2 11 10 PM
`BackingAddOnViewHolder`, `BackingAddOnViewHolderViewModel`

File: app/src/main/java/com/kickstarter/ui/viewholders/FrequentlyAskedQuestionsViewHolder.kt (hunk 1 above)
Comment (leighdouglas) [reply app/src/main/java/com/kickstarter/ui/viewholders/FrequentlyAskedQuestionsViewHolder.kt:39]: got it, updating now

File: app/src/test/java/com/kickstarter/viewmodels/FrequentlyAskedQuestionsViewHolderViewModelTest.kt [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import com.kickstarter.libs.utils.DateTimeUtils
+import com.kickstarter.libs.utils.extensions.addToDisposable
+import com.kickstarter.mock.factories.ProjectFaqFactory
+import io.reactivex.disposables.CompositeDisposable
+import io.reactivex.subscribers.TestSubscriber
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+
+class FrequentlyAskedQuestionsViewHolderViewModelTest : KSRobolectricTestCase() {
```
Comment (Arkariang) [inline app/src/test/java/com/kickstarter/viewmodels/FrequentlyAskedQuestionsViewHolderViewModelTest.kt:13]: 💖

File: app/src/main/java/com/kickstarter/ui/adapters/KSAdapter.java [hunk 4]
```diff
@@ -67,6 +67,7 @@ public void onViewDetachedFromWindow(final @NonNull KSViewHolder holder) {
     // to events, so ostensibly the view holder is being deallocated.
     if (!hasObservers()) {
       holder.lifecycleEvent(ActivityEvent.DESTROY);
+      holder.destroy();
```
Comment (Arkariang) [inline app/src/main/java/com/kickstarter/ui/adapters/KSAdapter.java:70]: 💖

**Review by Arkariang — APPROVED** [review]

**Review by Arkariang — APPROVED** [review]

**Pull request merged**